src/libraries/System.IO.FileSystem/tests/RandomAccess/NoBuffering.Windows.cs

```diff
@@ -242,6 +242,80 @@ public async Task ReadShouldReturnZeroForEndOfFile(bool asyncOperation, bool asy
             Assert.Equal(fileSize, total);
         }
 
+        [Fact]
+        public void ReadFromDiskAfterFlushToDisk()
+        {
+            // Save test file path so we can refer to the same file throughout the test.
+            string testFilePath = GetTestFilePath();
+
+            // Generate random bytes to write to file. NOTE: since we use unbuffered I/O,
+            // the byte count must be a multiple of the storage device's sector size. We
+            // could P/Invoke the DeviceIoControl() function to get the sector size by
+            // passing IOCTL_DISK_GET_DRIVE_GEOMETRY as a parameter, but for the sake of
+            // simplicity we will just use the system page size as a proxy. This works because
+            // both the system page size and the storage device's sector size are typically
+            // powers of 2 meaning the system page size (which is typically 4,096 bytes) will
+            // be a multiple of the sector size (which is typically 512 bytes) hence meeting
+            // the requirements for unbuffered I/O.
+            int byteCount = Random.Shared.Next(1, 10) * Environment.SystemPageSize;
```

Comment:
similarly to other test, I would prefer to make it a theory and provide multiple different values in explicit way as arguments (test cases)

Reply:
Fixed with [this commit](https://github.com/dotnet/runtime/pull/89100/commits/276f2c1706f60b26882c87acc2265c45bd8c7a14) 👍